app/app/explore/components/AddLiquidityModal.tsx

```diff
@@ -0,0 +1,274 @@
+  const { loading, error, addLiquidity } = useLiquidityPosition({ pool });
+  const [selectedTab] = useState("single");
+  const [assetAmount, setAssetAmount] = useState(0);
+  const [runeAmount, setRuneAmount] = useState(0);
+  const [txHash, setTxHash] = useState<string | null>(null);
+
+  // Initialize contract hooks
+  const poolViemAddress = pool.asset.split(".")[1].split("-")[1];
+  const tokenAddress = isERC20(pool.asset)
+    ? normalizeAddress(poolViemAddress)
+    : ""; // Set to empty string for non-ERC20 assets
+
+  // Use useContracts for ERC20 interaction
+  const {
+    balance: tokenBalance,
+    error: tokenError,
+    loadMetadata,
+  } = useContracts({
+    tokenAddress,
+    provider: wallet?.provider,
+  });
+
+  // Load token metadata on mount
+  useEffect(() => {
+    if (wallet?.provider) {
+      loadMetadata();
+    }
+  }, [wallet?.provider, loadMetadata]);
+
+  const getPercentage = (amount: number, max: number) => {
+    return max > 0 ? (amount / max) * 100 : 0;
+  };
+
+  const [runeBalance, setRuneBalance] = useState(0);
+  const getRuneBalance = useCallback(async () => {
+    if (!wallet?.address) return;
+    const { data: runeBalance } = await getBalance({
+      path: {
+        address: wallet.address,
+      },
+    });
+    return runeBalance;
+  }, [wallet]);
+
+  // Fetch RUNE balance on mount and every 10 seconds
+  useEffect(() => {
+    const fetchRuneBalance = async () => {
+      const balance: Balance | undefined = await getRuneBalance();
+      const amountStr: string = balance?.coins[0]?.amount || "0";
+      setRuneBalance(parseInt(amountStr));
+    };
+
+    fetchRuneBalance();
+    const intervalId = setInterval(fetchRuneBalance, 10000);
+    return () => clearInterval(intervalId);
+  }, [getRuneBalance]);
+
+  // Get formatted asset balance from tokenBalance
+  const assetBalance = useMemo(() => {
+    if (!tokenBalance?.formatted) return 0;
+    return Number(tokenBalance.formatted);
+  }, [tokenBalance]);
+
+  const currentAssetPercentage = useMemo(() => {
+    return getPercentage(assetAmount, assetBalance);
+  }, [assetAmount, assetBalance]);
+
+  const currentRunePercentage = useMemo(() => {
+    return getPercentage(runeAmount, runeBalance);
+  }, [runeAmount, runeBalance]);
+
+  const isCloseToPercentage = (
+    currentPercentage: number,
+    targetPercentage: number,
+  ) => {
+    const tolerance = 0.01;
+    return Math.abs(currentPercentage - targetPercentage) <= tolerance;
+  };
+
+  const handlePercentageClick = (
+    percentage: number,
+    isRune: boolean = false,
+  ) => {
+    if (isRune) {
+      const newRuneAmount = runeBalance * (percentage / 100);
+      setRuneAmount(newRuneAmount);
+    } else {
+      const newAssetAmount = assetBalance * (percentage / 100);
+      setAssetAmount(newAssetAmount);
+    }
+  };
+
+  const handleAddLiquidity = async () => {
+    if (!wallet?.address) {
+      alert("Please connect your wallet first");
```

Comment:
non-blocking but the button cta should be connect wallet if the wallet isn't already connected. once connected, the add liquidity modal should update to reflect the active wallet state.